packages/discord.js/src/structures/GuildMember.js

```diff
@@ -307,12 +307,12 @@ class GuildMember extends Base {
 
   /**
    * Sets the nickname for this member.
-   * @param {?string} nick The nickname for the guild member, or `null` if you want to reset their nickname
+   * @param {?string} nick The nickname for the guild member. Pass `null` to reset the nickname
    * @param {string} [reason] Reason for setting the nickname
    * @returns {Promise<GuildMember>}
    */
   setNickname(nick, reason) {
-    return this.edit({ nick, reason });
+    return this.id === this.client.user.id ? this.guild.members.editMe({ nick, reason }) : this.edit({ nick, reason });
```

Comment:
Is this the only thing specific to the clients guildMember? Just wondering if we may want a helper for this or an extended class like ClientUser (doubt)

Reply:
Yeah, if this is the only thing neither was worth it. `ClientGuildMember` would only be warranted with 4+ unique traits IMO